packages/testing/src/test/chai-retry-plugin.unit.ts

```diff
@@ -0,0 +1,295 @@
+        let attempts = 0;
+
+        const funcToRetry = () => {
+            attempts++;
+            if (attempts < 3) {
+                throw new Error('Failed');
+            }
+            return 'Success';
+        };
+
+        await expect(funcToRetry).to.retry({ retries: 5, delay: 10 });
+        expect(attempts).to.equal(3);
+    });
+
+    it('should retry a function that always fails and throw an error', async () => {
+        let attempts = 0;
+
+        const funcToRetry = () => {
+            attempts++;
+            throw new Error('Failed');
+        };
+
+        try {
+            await expect(funcToRetry).to.retry({ retries: 3 });
+
+            throw new Error('This should not be called');
+        } catch (error) {
+            expect(error).to.be.an.instanceOf(Error);
+            expect((error as Error).message).to.includes('Limit of 3 retries');
+            expect(attempts).to.equal(3);
+        }
+    });
+
+    describe('options should work correctly:', () => {
+        it('timeout after the specified duration', async () => {
+            const funcToRetry = async () => {
+                await new Promise((resolve) => setTimeout(resolve, 500));
+                return 'Success';
+            };
+
+            await expect(funcToRetry)
+                .retry({ retries: 2, timeout: 250 })
+                .then(
+                    () => {
+                        throw new Error('This should not be called');
+                    },
+                    (error: Error) => {
+                        expect(error).to.be.an.instanceOf(Error);
+                        expect(error.message).to.equal('timed out after 250ms');
+                    }
+                );
+        });
+
+        it('throw an error when retries limit exceeded', async () => {
+            let attempts = 0;
+
+            const funcToRetry = () => {
+                attempts++;
+
+                return attempts;
+            };
+
+            await expect(funcToRetry)
+                .retry({ retries: 10 })
+                .to.be.above(100)
+                .then(
+                    () => {
+                        throw new Error('This should not be called');
+                    },
+                    (error: Error) => {
+                        expect(error).to.be.an.instanceOf(Error);
+                        expect(error.message).to.equal('Limit of 10 retries exceeded!');
+                    }
+                );
+        });
+
+        it('should retry a function that succeeds and assert the result with getter property', async () => {
+            let attempts = 0;
+
+            const funcToRetry = () => {
+                attempts++;
+                if (attempts < 3) {
+                    throw new Error('Failed');
+                }
+                return { success: true };
+            };
+
+            await expect(funcToRetry).retry({ retries: 5, delay: 10 }).and.have.property('success').and.be.true;
```

Comment:
if we have specific tests for retry options working correctly, let's not pass options in other tests